addons/sales_team/tests/test_sales_team_membership.py

```diff
@@ -303,3 +303,26 @@ def test_users_sale_team_id(self):
         self.sales_team_1_m1.unlink()
         self.assertEqual(self.user_sales_leads.crm_team_ids, self.new_team)
         self.assertEqual(self.user_sales_leads.sale_team_id, self.new_team)
+
+    def test_sales_team_member_search(self):
+        # create a team
+        new_team = self.env['crm.team'].create({'name': 'Test Team'})
+        # create a res.partner
+        partner = self.env['res.partner'].create({'name': 'Test Partner'})
+
+        # add the new_team as partner team_id value
+        partner.write({'team_id': new_team.id})
+
+        # add the current user to the sales team
+        new_team.write({'member_ids': [(4, self.env.uid)]})
+
+        # remove the current user from the sales team
+        new_team.write({'member_ids': [(3, self.env.uid)]})
```

Comment:
The thing is that we don't issue an unlink command (3) when removing from the m2m in the front-end, we issue a set command (6) with no value. Which would just be `new_team.member_ids = []` in the back-end.
This then calls `_inverse_member_ids` which archives the team member :)

So you can just use regular assignments instead of commands and it should work

Reply:
hi @thiryr ,
updated the test case, now the test case if failing when the fix is not there and after the fix test is getting passed also,
thanks